src/Bicep.Core/Semantics/ParamAssignmentSymbolCollectVisitor.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Bicep.Core.Syntax;
+using Bicep.Core.Workspaces;
+
+namespace Bicep.Core.Semantics
+{
+    public sealed class ParamAssignmentSymbolCollectVisitor : SyntaxVisitor
```

Comment:
Is it worth naming this something to indicate that it's the params-file equivalent of the `DeclarationVisitor`?